modepy/modes.py

```diff
@@ -673,6 +709,11 @@ def grad_tensor_product_basis(dims, basis_1d, grad_basis_1d):
 
     .. versionadded:: 2020.2
     """
+    warn("grad_tensor_product_basis is deprecated. "
+            "Use TensorProductBasis instead. "
```

Comment:
```suggestion
            "Use basis_for_shape instead. "
```
Same as for `tensor_product_basis`.

Reply:
7f4e1a1 and 347027b